### ISSUE TITLE
Install manual for less experienced user 

### ISSUE THREAD

**Issue by thundergreen** [siacs/HttpUploadComponent#53]
Hi guys. It would be highly appreciated to make an install tutorial for less experienced user.like this the dev-chan would be less full of questions about this. Maybe this would be possible?

A big fan of your whole project

**Comment by knoy** [comment]
I agree.. this is a very confusing repo.

**Comment by jagdishkk1989** [comment]
I also agree. I am struggling since last week. How to test the HttpUploadComponent on localhost? Really very confusing..:(

**Comment by thundergreen** [comment]
If you have Ejabberd running I can help u with the httpupload component of
Ejabberd .I got siacs httpuploadcomponent one time running but don't
remember how.with Ejabberd it also works pretty good .add me to your
contact list if you want : thorsten@emevth.no-ip.biz

**Comment by jagdishkk1989** [comment]
Thank you so much!! I am installing the Ejabberd then let you know. I was trying on openfire. Thanks again!!

**Comment by thundergreen** [comment]
You are welcome ;)

**Comment by gakwaya** [comment]
@thundergreen  ,Hello ,did you get this module to work with ejabberd? I am also looking for a simple guide on how how to get it running . More on my problem here : https://github.com/siacs/HttpUploadComponent/issues/56
Thanks.

**Comment by jagdishkk1989** [comment]
@thundergreen, No, I have installed for openfire on live server because openfire admin console is very user friendly.

**Comment by gakwaya** [comment]
Could you elaborate on how you installed ? or point me to a useful piece of doc? I am familiar with openfire .If I get it to work with openfire I can port the procedure to ejabberd.

**Comment by jagdishkk1989** [comment]
Yes, You need to download the HTTP component and install python as per
steps.

- Define the external component port and default secret in openfire using
admin console in server setting external component settings.
- Change the in the config.yml  and define the configration. Please check
the ReadMe of http  upload component.
- Upload http upload component via ftp on your server, where openfire also
installed and restart. If component connected then will be show in openfire
connected component.



-

On Tue, Jan 19, 2016 at 11:06 AM, Daniel Gakwaya <notifications@github.com>
wrote:

> Could you elaborate on how you installed ? or point me to a useful piece
> of doc? I am familiar with openfire .If I get it to work with openfire I
> can port the procedure to ejabberd.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/siacs/HttpUploadComponent/issues/53#issuecomment-172743887>
> .
>



-- 
Thanks & Regards,
Jagdish

**Comment by mt403** [comment]
Some short instructions for the people struggling with the installation.
I assume you are running Debian/Raspbian.

### install python 3.4 and virutalenv
`sudo apt-get install python3.4 python-virtualenv`

### clone the repository
```
git clone https://github.com/siacs/HttpUploadComponent httpupload
cd httpupload

```
### create virtualenv and install depencies
```
virtualenv -p /usr/bin/python3.4 venv
source venv/bin/activate
pip install -r requirements.txt
```

For more info see: [http://docs.python-guide.org/en/latest/dev/virtualenvs](http://docs.python-guide.org/en/latest/dev/virtualenvs)

### create config
`cp config.example.yml config.yml`

Open the config file with your favourite editor and change to your needs.

### use it
At this point you should be able to start the server with:
`python httpupload/server.py`

After you're done testing exit the virtualenv with
`deactivate`

### systemd service file
Since httpuploadcomponent is running in a virtualenv you will a need a slightly different
service file then the one provided in the contrib directory.

```
[Unit]
Description=XMPP component for HTTP file uploads
After=network.target

[Service]
Type=simple
Restart=always

Environment=VIRTUAL_ENV=/opt/httpupload/venv
Environment=PATH=$VIRTUAL_ENV/bin:$PATH

ExecStart=/opt/httpupload/venv/bin/python /opt/httpupload/httpupload/server.py
WorkingDirectory=/opt/httpupload

User=www-data

[Install]
WantedBy=multi-user.target
```

If you cloned the repository somewhere else be sure to adopt to your path.

I assume you're are going to proxy file uploads/downloads through a web server hence
the www-data user.

This should leave you with a working installation of httpuploadcomponent.
You still will have to tell your xmpp server about the component and configure
your web server.

I'm no expert so this howto could contain flaws if you find some please post your corrections here.

**Comment by KlausHans** [comment]
Thank you. Unfortunately, i could not get it to work. The problem is, that after
python httpupload/server.py
it throws serveral errors, because python can't find several imported libs. Some of them i could install manually, but now i am hanging at http.server. I installed the http module, but it still says it could not find "server".
So, witch pythons librarys have i to install exactly?
I'm on Debian 8.3

**Comment by kriztan** [comment]
Take a look into https://github.com/siacs/HttpUploadComponent/blob/master/requirements.txt
There is a list with required libs.

**Comment by KlausHans** [comment]
Thank you very much for that ultra fast reply! I will have a look.

**Comment by ageekhere** [comment]
Hi all, I am having an issue running the script with openfire
my config
-----------------
component_jid: 192.168.1.5 
component_secret: mysecret
component_port: 5275

storage_path : /myfiles

whitelist:
  - 192.168.1.5
max_file_size: 20971520 #20MiB

http_address: 127.0.0.1 #use 0.0.0.0 if you don't want to use a proxy
http_port: 8081
#http_keyfile: /etc/ssl/private/your.key
#http_certfile: /etc/ssl/certs/your.crt

get_url : http://192.168.1.5:8081
put_url : http://192.168.1.5:8081

expire_interval: 82800  #time in secs between expiry runs (82800 secs = 23 hours). set to '0' to disable
expire_maxage: 2592000  #files older than this (in secs) get deleted by expiry runs (2592000 = 30 days)
user_quota_hard: 104857600   #100MiB. set to '0' to disable rejection on uploads over hard quota
user_quota_soft: 78643200    #75MiB. set to '0' to disable deletion of old uploads over soft quota an expiry runs
allow_web_clients: true #answer OPTIONS requests to allow web clients to upload files

When i run
python3 httpupload/server.py
The ERROR
----------------------
2016-03-22 11:04:53,320  DEBUG    Expire run finished in 0.000728s
2016-03-22 11:04:53,340  DEBUG    setting jid to 192.168.1.5
2016-03-22 11:04:53,355  DEBUG    Loaded Plugin (XEP-0030) Service Discovery
2016-03-22 11:04:53,375  DEBUG    Loaded Plugin (XEP-0999) upload files via http
2016-03-22 11:04:53,376  DEBUG    Connecting to localhost:5275
2016-03-22 11:04:53,377  DEBUG    Connecting to localhost:5275
2016-03-22 11:04:53,380  DEBUG     ==== TRANSITION disconnected -> connected
2016-03-22 11:04:53,383  DEBUG    Starting HANDLER THREAD
2016-03-22 11:04:53,385  DEBUG    Loading event runner
2016-03-22 11:04:53,387  DEBUG    SEND (IMMED): <stream:stream xmlns="jabber:component:accept" xmlns:stream="http://etherx.jabber.org/streams" to='192.168.1.5'>
connected
2016-03-22 11:04:53,415  DEBUG    SEND (IMMED): <handshake xmlns="jabber:component:accept">f2e62db55c793ab7e424391dd8399b2616fafc98</handshake>
2016-03-22 11:04:53,416  DEBUG    RECV: <error xmlns="http://etherx.jabber.org/streams"><internal-server-error xmlns="urn:ietf:params:xml:ns:xmpp-streams" /></error>
2016-03-22 11:04:53,420  ERROR    Error reading from XML stream.
2016-03-22 11:04:53,421  DEBUG    Ending read XML loop
2016-03-22 11:04:53,423  DEBUG    reconnecting...
2016-03-22 11:04:53,424  DEBUG    SEND (IMMED): </stream:stream>
2016-03-22 11:04:53,425  WARNING  Failed to send </stream:stream>
2016-03-22 11:04:53,427  DEBUG    Could not acquire lock

Any Ideas?

**Comment by mt403** [comment]
I'm not sure you are allowed to use ip addresses instead of domain names in xmpp.
At least you have to give the component a name and specify the jid like this:

**component_jid: upload@192.168.1.5 **

I would really use a domain even if you are testing this internally to avoid further errors.

**Comment by ageekhere** [comment]
Hi. I normally use a DDNS however I was getting the same error so I thought I would try just the ip of the server
--------------------------
component_jid: upload.myDDNSgoesHERE.ddns.net.tld  #NOTE I am guessing that the address goes between the upload and tld (tried with and without)
component_secret: mysecret #added it to openfire
component_port: 5275 #port for openfire External Components
storage_path : ./ #will change later
whitelist:
  - myDDNSgoesHERE.ddns.net.tld #guessing .tld is needed here
max_file_size: 20971520 #20MiB

http_address: 127.0.0.1 #use 0.0.0.0 if you don't want to use a proxy
http_port: 8081 

#http_keyfile: /etc/ssl/private/your.key
#http_certfile: /etc/ssl/certs/your.crt

get_url : http://upload.myDDNSgoesHERE.ddns.net.tld:8081
put_url : http://upload.myDDNSgoesHERE.ddns.net.tld:8081

expire_interval: 82800  #time in secs between expiry runs (82800 secs = 23 hours). set to '0' to disable
expire_maxage: 2592000  #files older than this (in secs) get deleted by expiry runs (2592000 = 30 days)
user_quota_hard: 104857600   #100MiB. set to '0' to disable rejection on uploads over hard quota
user_quota_soft: 78643200    #75MiB. set to '0' to disable deletion of old uploads over soft quota an expiry runs
allow_web_clients: true #answer OPTIONS requests to allow web clients to upload files

ERROR
--
2016-03-22 19:11:25,017  DEBUG    Expire run finished in 0.004998s
2016-03-22 19:11:25,037  DEBUG    setting jid to upload.myDDNSgoesHERE.ddns.net.tld
2016-03-22 19:11:25,052  DEBUG    Loaded Plugin (XEP-0030) Service Discovery
2016-03-22 19:11:25,057  DEBUG    Loaded Plugin (XEP-0999) upload files via http
2016-03-22 19:11:25,058  DEBUG    Connecting to localhost:5275
2016-03-22 19:11:25,059  DEBUG    Connecting to localhost:5275
2016-03-22 19:11:25,061  DEBUG     ==== TRANSITION disconnected -> connected
2016-03-22 19:11:25,063  DEBUG    Starting HANDLER THREAD
2016-03-22 19:11:25,065  DEBUG    Loading event runner
2016-03-22 19:11:25,067  DEBUG    SEND (IMMED): <stream:stream xmlns="jabber:component:accept" xmlns:stream="http://etherx.jabber.org/streams" to='upload.myDDNSgoesHERE.ddns.net.tld'>
connected
2016-03-22 19:11:25,088  DEBUG    SEND (IMMED): <handshake xmlns="jabber:component:accept">4d59b596b5baeb3437ed23710d790a21a79f53b9</handshake>
2016-03-22 19:11:25,090  DEBUG    RECV: <error xmlns="http://etherx.jabber.org/streams"><internal-server-error xmlns="urn:ietf:params:xml:ns:xmpp-streams" /></error>
2016-03-22 19:11:25,092  ERROR    Error reading from XML stream.
2016-03-22 19:11:25,094  DEBUG    Ending read XML loop
2016-03-22 19:11:25,094  DEBUG    reconnecting...
2016-03-22 19:11:25,095  DEBUG    SEND (IMMED): </stream:stream>
2016-03-22 19:11:25,096  WARNING  Failed to send </stream:stream>
2016-03-22 19:11:25,098  DEBUG    Could not acquire lock

Where am i going wrong?

**Comment by mt403** [comment]
Sorry, got confused myself. The jid needs to look like this (assuming your domain is you.ddns.net):

**upload.you.ddns.net**

Basically you just append upload. to your domain.

Did you specify the component name and the component secret in openfire ?
Can you check the server logs and see if the component authenticates with the server ?

**Comment by ageekhere** [comment]
Ok Worked it out you do not need ,tld
Also in openfire the Default shared secret had a space in it. Anyway it now connects, now to see if it works with monal im and jitsi

**Comment by Aravinthu** [comment]
Please help some one

> My Config

component_jid: bdata.aravinth.co
component_secret: admin
component_port: 5347

storage_path : ./

whitelist:
  - bdata.aravinth.co

max_file_size: 20971520 #20MiB

http_address: 127.0.0.1 #use 0.0.0.0 if you don't want to use a proxy
http_port: 8080
#http_keyfile: /etc/ssl/private/your.key
#http_certfile: /etc/ssl/certs/your.crt

get_url : http://bdata.aravinth.co:8080
put_url : http://bdata.aravinth.co:8080

> The Error

DEBUG    Expire run finished in 0.000413s
DEBUG    Loaded Plugin: XEP-0030: Service Discovery
DEBUG    Loaded Plugin: upload files via http
DEBUG    Connecting to localhost:5347
DEBUG    DNS: Querying localhost for AAAA records.
DEBUG    DNS: Error retreiving AAAA address info for localhost.
DEBUG    DNS: Querying localhost for A records.
DEBUG    Connecting to [::1]:5347
DEBUG    Event triggered: socket_error
ERROR    Could not connect to [::1]:5347. Socket Error #111: Connection refused
DEBUG    Connecting to 127.0.0.1:5347
DEBUG    Event triggered: socket_error
ERROR    Could not connect to 127.0.0.1:5347. Socket Error #111: Connection refused
DEBUG    Connecting to 127.0.0.1:5347
DEBUG    Event triggered: socket_error
ERROR    Could not connect to 127.0.0.1:5347. Socket Error #111: Connection refused
DEBUG    No remaining DNS records to try.
DEBUG    Waiting 1.93296282304 seconds before connecting.

**Comment by mt403** [comment]
Several things:

Does your xmpp server know about the component ?

Is bdata.aravinth.co the component name or the xmpp domain ?
If it is the component name then you have to change the whitelist setting.

It seems that you don't want to proxy your uploads, so http_address should be 0.0.0.0.

**Comment by Aravinthu** [comment]
@mt403 Running now i just changed the port to 5275, is this correct.

DEBUG    Expire run finished in 0.000584s
DEBUG    Loaded Plugin: XEP-0030: Service Discovery
DEBUG    Loaded Plugin: upload files via http
DEBUG    Connecting to localhost:5275
DEBUG    DNS: Querying localhost for AAAA records.
DEBUG    DNS: Error retreiving AAAA address info for localhost.
DEBUG    DNS: Querying localhost for A records.
DEBUG    Connecting to [::1]:5275
DEBUG    Event triggered: connected
DEBUG     ==== TRANSITION disconnected -> connected
DEBUG    Starting HANDLER THREAD
DEBUG    Loading event runner connected
DEBUG    SEND (IMMED): <stream:stream xmlns="jabber:component:accept" xmlns:stream="http://etherx.jabber.org/streams" to='bdata.aravinth.co'>
DEBUG    RECV: <stream:stream id="2vpa2qcoqc" from="bdata.aravinth.co.bigdata">
DEBUG    SEND (IMMED): <handshake xmlns="jabber:component:accept">c2c8b9e59253ac2243933bf642dccfccf4dc9681</handshake>
DEBUG    RECV: <handshake />
DEBUG    Event triggered: session_bind
DEBUG    Event triggered: session_start
DEBUG    RECV: <iq id="907-18" type="get" to="bdata.aravinth.co.bigdata" from="component.bigdata"><query xmlns="http://jabber.org/protocol/disco#info" /></iq>
DEBUG    Received disco info query from <component.bigdata> to <bdata.aravinth.co.bigdata>.
DEBUG    No identity found for this entity. Using default component identity.
DEBUG    No features found for this entity. Using default disco#info feature.
DEBUG    SEND: <iq id="907-18" type="result" to="component.bigdata" from="bdata.aravinth.co.bigdata"><query xmlns="http://jabber.org/protocol/disco#info"><identity category="component" type="generic" /><feature var="http://jabber.org/protocol/disco#info" /></query></iq>
DEBUG    Scheduled event: Session timeout check: ()

**Comment by mt403** [comment]
It seems to be connected. Try uploading something, if it fails check the stuff i pointed out above.

**Comment by Aravinthu** [comment]
@mt403 default port 5347 to 5275, ok now running but i can't see the attachment icon in my conversation 
android client.

**Comment by mt403** [comment]
Goto your account settings and check if it is listed in server info, if not reconnect and wait some time. If that does not help check the xmpp server logs to verify the component is connected.

**Comment by thundergreen** [comment]
Which Server are u running? Ejabberd has built in httpupload

Thorsten Fröhlich
Am 10.05.2016 11:26 schrieb "mt403" <notifications@github.com>:

**Comment by stevenroose** [comment]
Why has this become a general "my setups gives this error" thread? Someone who knows how to do it should put install instructions. Both the trivial options `python setup.py` and `make` fail, so there should be instructions.

**Comment by thundergreen** [comment]
Hi Steve. Why not using built in httpupload in ejabberd?

Thorsten Fröhlich
Am 30.05.2016 22:10 schrieb "Steven Roose" <notifications@github.com>:

**Comment by stevenroose** [comment]
@thundergreen I use Prosody. And I was told that this standalone component would give better performance :)

**Comment by thundergreen** [comment]
Well sorry. I'm not familiar with prosody and its  configuration
possibilities..

Thorsten Fröhlich
Am 31.05.2016 10:31 schrieb "Steven Roose" <notifications@github.com>:

**Comment by tahatauqeer** [comment]
@thundergreen could you please tell me how to configure ejabberd on windows?

**Comment by thundergreen** [comment]
Ejabberd is mit suppose to run on windows

Thorsten Fröhlich

Am 09.10.2016 17:32 schrieb "tahatauqeer" <notifications@github.com>:

**Comment by tahatauqeer** [comment]
@thundergreen i didn't understand what do you mean by it "Ejabberd is mit suppose to run on windows"

**Comment by thundergreen** [comment]
U asked me how to get Ejabberd running under windows. I answered that
Ejabberd is not supposed to run under windows

Thorsten Fröhlich

Am 10.10.2016 07:01 schrieb "tahatauqeer" <notifications@github.com>:

**Comment by tahatauqeer** [comment]
@thundergreen can you tell me how to add HTTPFileUploadComponent to openfire server? Or suggest me a good tutorial to do this. Thanks

**Comment by tahatauqeer** [comment]
@ageekhere  kindly tell me how to add this component to openfire?

**Comment by thundergreen** [comment]
Im afraid this won't work with openfire.chevk out my Tutorial hownto
install Ejabberd https://emevth.no-ip.biz/Tutorial

Thorsten Fröhlich

Am 11.10.2016 09:34 schrieb "tahatauqeer" <notifications@github.com>:

**Comment by tahatauqeer** [comment]
@thundergreen Error 404
Woops. Looks like this page doesn't exist.

**Comment by tahatauqeer** [comment]
@thundergreen sorry. i get it. Thank you so much :)

**Comment by thundergreen** [comment]
Sorry ..wrong link

It is: https://emevth.no-ip.biz/tutorials

Thorsten Fröhlich

Am 15.10.2016 11:01 schrieb "tahatauqeer" <notifications@github.com>:
